libpromises/evalfunction.c

```diff
@@ -949,6 +949,15 @@ static StringSet *VariablesMatching(const EvalContext *ctx, VariableTableIterato
     Variable *v = NULL;
     while ((v = VariableTableIteratorNext(iter)))
     {
+        if (strchr(v->ref->lval, '#') != NULL)
```

Comment:
Can `v->ref` be `NULL`?

Reply:
No. We are assuming this in many other places (not only) in this function.